src/notebooks/outputs/errorRendererComms.ts

```diff
@@ -29,33 +39,36 @@ export class ErrorRendererCommunicationHandler implements IExtensionSyncActivati
 
     activate(): void {
         const messageChannel = notebooks.createRendererMessaging('jupyter-error-renderer');
-        this.disposables.push(
-            messageChannel.onDidReceiveMessage(async (e) => {
-                const message = e.message;
-                if (message.message === InteractiveWindowMessages.OpenLink) {
-                    const href = message.payload;
-                    if (href.startsWith('file')) {
-                        await this.openFile(href);
-                    } else if (href.startsWith('vscode-notebook-cell')) {
-                        await this.openCell(href);
-                    } else if (href.startsWith('https://command:') || href.startsWith('command:')) {
-                        const temp: string = href.startsWith('https://command:')
-                            ? href.split(':')[2]
-                            : href.split(':')[1];
-                        const params: string[] = temp.includes('/?') ? temp.split('/?')[1].split(',') : [];
-                        let command = temp.split('/?')[0];
-                        if (command.endsWith('/')) {
-                            command = command.substring(0, command.length - 1);
-                        }
-                        if (linkCommandAllowList.includes(command)) {
-                            await commands.executeCommand(command, params);
-                        }
-                    } else {
-                        this.applicationShell.openUrl(href);
-                    }
+        messageChannel.onDidReceiveMessage(this.onDidReceiveMessage, this, this.disposables);
+    }
+
+    // Public for testing
+    public async onDidReceiveMessage(e: {
```

Comment:
You mentioned for testing, but honestly I like this better just as a general thing. Lambdas always look funny to me when they get really long.